recordset/recordset.js

```diff
@@ -89,193 +82,194 @@ angular.module('recordset', ['ermrestjs', 'chaise.navbar', 'chaise.utils', 'chai
+        });
     };
 
-    $scope.sortby = function(column) {
-        if (recordsetModel.sortby !== column) {
-            recordsetModel.sortby = column;
-            recordsetModel.sortOrder = "asc";
-            $scope.sort();
+    $scope.permalink = function() {
+
+        // before run, use window location
+        if (!$rootScope.reference) {
+            return $window.location.href;
         }
 
-    };
+        var url = context.mainURI;
 
-    $scope.toggleSortOrder = function () {
-        recordsetModel.sortOrder = (recordsetModel.sortOrder === 'asc' ? recordsetModel.sortOrder = 'desc' : recordsetModel.sortOrder = 'asc');
-        $scope.sort();
-    };
+        // add sort modifier
+        if ($rootScope.reference.location.sort)
+            url = url + $rootScope.reference.location.sort;
 
-    $scope.permalink = function() {
-        var url = $window.location.href.replace($window.location.hash, ''); // everything before #
-        url = url + "#" + UriUtils.fixedEncodeURIComponent(context.catalogID) + "/" +
-            (context.schemaName !== '' ? UriUtils.fixedEncodeURIComponent(context.schemaName) + ":" : "") +
-            UriUtils.fixedEncodeURIComponent(context.tableName);
+        // add paging modifier
+        if ($rootScope.reference.location.paging)
+            url = url + $rootScope.reference.location.paging;
 
-        if (recordsetModel.filter !== null) {
-            url = url + "/" + recordsetModel.filter.toUri();
-        }
+        url = url + "?limit=" + pageInfo.pageLimit;
 
-        if (recordsetModel.sortby !== null) {
-            url = url + "@sort(" + UriUtils.fixedEncodeURIComponent(recordsetModel.sortby);
-            if (recordsetModel.sortOrder === "desc") {
-                url = url + "::desc::";
-            }
-            url = url + ")";
-        }
         return url;
     };
 
     $scope.before = function() {
 
-        if (pageInfo.recordStart > 1) { // not on page 1
+        var previous = recordsetModel.page.previous;
+        if (previous) {
 
             pageInfo.loading = true;
 
             // disable buttons while loading
             pageInfo.previousButtonDisabled = true;
             pageInfo.nextButtonDisabled = true;
 
-            recordsetModel.rowset.before().then(function (rowset) {
-                console.log(rowset);
-                $window.scrollTo(0, 0);
-                recordsetModel.rowset = rowset;
-                pageInfo.recordStart -= pageInfo.pageLimit;
-                pageInfo.recordEnd = pageInfo.recordStart + rowset.length() -1;
+            $rootScope.reference = previous; // TODO contextualize for recordset
+            $log.info("Reference:", $rootScope.reference);
 
-                pageInfo.loading = false;
+            recordsetModel.columns = $rootScope.reference.columns.map(function (column, index, array) {
+                return {"name": column.name, "displayname": column.displayname};
+            });
```

Comment:
Why not just `recordsetModel.columns = $rootScope.reference.columns`? Is it something like you just only need the name and displayname properties of each column?